sdk/core/core-rest-pipeline/src/util/helpers.ts

```diff
@@ -27,10 +27,14 @@ export function delay<T>(
     let timer: ReturnType<typeof setTimeout> | undefined = undefined;
     let onAborted: (() => void) | undefined = undefined;
 
+    const abortReason =
+      // use the signal's reason if available
+      options?.abortSignal?.reason ??
+      // fallback to legacy code
+      options?.abortErrorMsg ??
```

Comment:
What I would like to open for discussion is whether `options?.abortErrorMsg` takes precedence over the abortSignal's reason or whether we should always try to bubble up abortSignal reason first.